### PR TITLE
Initial version with gRPC + gRPC-gateway + wire + loggo + cli

### PR DESCRIPTION
**Pull request by cocotyty**
Implementation of GET /v1/api/objects/:type_name

### CODE REVIEW

File: cmd/server/wire_gen.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Code generated by Wire. DO NOT EDIT.
```
Comment (sunxiaoguang) [inline cmd/server/wire_gen.go:1]: Duplicated InitServer function in wire.go and wire_gen.go

File: cmd/server/main.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	ctx := signals.SignalHandler(context.Background())
+	// Register gRPC server endpoint
+	// Note: Make sure the gRPC server is running properly and accessible
+
+	app := cli.NewApp()
+
+	app.Flags = []cli.Flag{
+		&cli.StringFlag{
+			Name:    FlagLogConf,
+			Usage:   "config logger level,see http://github.com/juju/loggo",
```
Comment (sunxiaoguang) [inline cmd/server/main.go]: It looks weird without space between comma and see

File: cmd/server/main.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+			EnvVars: []string{"LOG_CONF"},
+			Value:   "<root>=INFO",
+		},
+		&cli.StringFlag{
```
Comment (sunxiaoguang) [inline cmd/server/main.go]: we can use github.com/soheilhy/cmux to serve request for both gRPC and HTTP on the same port.

File: pkg/api/v1/object.proto [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+import "google/protobuf/timestamp.proto";
+import "google/api/annotations.proto";
+
+message ObjectListRequest {
+  string type = 1;
+  ObjectView view = 2;
+  string query = 3;
+  string page_token = 4;
+  int32 pageSize = 5;
```
Comment (sunxiaoguang) [inline pkg/api/v1/object.proto]: snake case should be used instead

File: cmd/server/main.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+			EnvVars: []string{"GRPC_ADDR"},
+			Value:   ":9090",
+		},
+		&cli.StringFlag{
+			Name:    FlagHTTPAddr,
+			Usage:   "HTTP protocol endpoint address",
+			EnvVars: []string{"HTTP_ADDR"},
+			Value:   ":8080",
+		},
+		&cli.StringFlag{
```
Comment (sunxiaoguang) [inline cmd/server/main.go:65]: Can we add a flag to turn on http pprof endpoints?

File: cmd/server/wire_gen.go (hunk 1 above)
Comment (cocotyty) [reply cmd/server/wire_gen.go:1]: see here: https://github.com/google/wire/tree/master/_tutorial

File: pkg/api/v1/object.proto [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+enum ValueType {
+  STRING  = 0;
+  INTEGER = 1;
+  DOUBLE  = 2;
+  BOOLEAN = 3;
+}
+
+message ObjectListResponse {
+           string kind            = 1;
+  repeated Object objects         = 2;
```
Comment (sunxiaoguang) [inline pkg/api/v1/object.proto]: Indention

File: pkg/api/v1/object.proto [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+message Object {
+  string type = 1;
+  string name = 2;
+  string description = 3;
+  string status = 4;
+  string state = 5;
+  google.protobuf.Timestamp create_time = 6;
+  map<string,ObjectMeta> metas = 7;
```
Comment (sunxiaoguang) [inline pkg/api/v1/object.proto]: Add a space after comma

File: sql/ddl.sql [hunk 8]
```diff
@@ -68,7 +68,7 @@ CREATE TABLE object_meta (
   type_id SMALLINT UNSIGNED NOT NULL,
   name VARCHAR(256) NOT NULL,
   value_type TINYINT UNSIGNED NOT NULL DEFAULT 1 COMMENT '1: STRING 2: INTEGER, 3: DOUBLE, 4: BOOLEAN',
-  description VARCHAR(1024),
+  description VARCHAR(1024)  NOT NULL,
```
Comment (sunxiaoguang) [inline sql/ddl.sql]: redundant spaces

File: sql/ddl.sql [hunk 9]
```diff
@@ -50,10 +50,10 @@ CREATE TABLE object (
 CREATE TABLE deleted_object (
   id BIGINT UNSIGNED NOT NULL,
   type_id SMALLINT UNSIGNED NOT NULL,
-  name VARCHAR(256) NOT NULL,
+  name VARCHAR(256)  NOT NULL,
   version BIGINT NOT NULL,
   relation_version BIGINT NOT NULL DEFAULT 0,
-  description VARCHAR(1024),
+  description VARCHAR(1024)  NOT NULL,
```
Comment (sunxiaoguang) [inline sql/ddl.sql]: Redundant spaces

**Review by sunxiaoguang — COMMENTED** [review]
LGTM

**Review by sunxiaoguang — DISMISSED** [review]

**Review by sunxiaoguang — APPROVED** [review]

**Pull request merged**